### ISSUE TITLE
Add a way to change ocamlformat configuration using attributes. 

### ISSUE THREAD

**Issue by hhugo** [ocaml-ppx/ocamlformat#272]
```ocaml
[@@@ocamlformat "extension-point: preserve"]

let x =
  if b then e else " " [@ocamlformat "escape-strings: preserve"]
[@@ocamlformat "if-then-else: keyword-first"]
```

In particular, it would be very useful in conjunction with #271 when formatting code using metaquot.

**Comment by jberdine** [comment]
This could also be an implementation strategy for resolving #263 .

Considering the current way that the `Conf.t` value is threaded through essentially all the formatting code, I think that this should not be difficult to implement.

**Comment by jberdine** [comment]
implemented in #286
